### ISSUE TITLE
Crash on exit (OpenGL 2.1)

### ISSUE THREAD

**Issue by deveee** [supertuxkart/stk-code#2322]
This probably occurs only in legacy engine (OpenGL 2.1). I got this during testing STK on Virtualbox, when I was closing the game using escape key:
```
Assertion failed: m_has_been_initialised, file E:\linux\supertuxkart-git\src\gra
phics\shaders.cpp, line 144
```

**Comment by auriamg** [comment]
Setting as minor because it happens 1) in an old OpenGL 2) in a VM 3) when closing.
All of these are factors that make this bug less important :) In part because OpenGL 2 is old, because VMs tend to have driver bugs that don't happen on real PCs, and also bugs when closing are less important since they don't prevent playing

**Comment by deveee** [comment]
The crash is fixed now (commit 11f6084fc52e93b335ffe1ddaa926a97af62a8f4), but still old renderer is not working. It crashes during creating skybox with similar m_has_been_initialized assertion. I will create separate bug report to avoid a mess.
